### PR TITLE
feat: Namespace selection for deploy to kubernetes

### PR DESCRIPTION
**Pull request by elervik**
### What does this PR do?
Adds a dropdown menu, populated with namespaces. Defaults to `default`namespace, if unable to get any from kube_client module.

If the user have set `kubectl config set-context --current --namespace=NAMESPACE` it should honor that as the chosen value.

Also adds a `<div>` around the "deploy" button to get some spacing between the menus.

### Screenshot/screencast of this PR
deploy1

deploy2

### What issues does this PR fix or reference?

Couldnt find any on this, but it delivers on the comment at
https://github.com/containers/podman-desktop/blob/main/packages/renderer/src/lib/pod/DeployPodToKube.svelte#L43

### How to test this PR?

Create a pod > Deploy

Signed-off-by: Endre Lervik <endrel@met.no>

### CODE REVIEW

File: packages/renderer/src/lib/pod/DeployPodToKube.svelte [hunk 1]
```diff
@@ -40,9 +41,14 @@ onMount(async () => {
   // grab current namespace
   currentNamespace = await window.kubernetesGetCurrentNamespace();
 
+  // check that the variable is set to a value, otherwise set to default namespace
+  if (!currentNamespace) {
+    currentNamespace = 'default';
```
Comment (cdrage) [inline packages/renderer/src/lib/pod/DeployPodToKube.svelte:46]: Yeah, you're right. As the function:

```
 getCurrentNamespace(): string | undefined {
    return this.currrentNamespace;
  }
 ```
 
 Would return undefined. This LGTM. From my testing it works. 
 
 Maybe put `default` as a const at the top of the file?

**Review by cdrage — CHANGES_REQUESTED** [review]
Thanks for the first contribution!! 🥂 

Looks like you committed `yarn.lock` by mistake. 

Other than one small change, this LGTM. Looked at the styling as well and that works UI/UX wise.

File: packages/extension-api/package.json [hunk 2]
```diff
@@ -16,6 +16,5 @@
     "build": "",
     "watch": "",
     "publish:next": "yarn publish  --registry=https://registry.npmjs.org/ --no-git-tag-version --new-version 0.0.1-\"$(date +%s)\""
-  },
-  "devDependencies": {}
```
Comment (cdrage) [inline packages/extension-api/package.json]: ? change by mistake?

**Review by cdrage — CHANGES_REQUESTED** [review]

**Comment by cdrage** [comment]
You'll have to sign the commit (there's no description in the commit)

and run `yarn format:fix`

Thanks again!

File: packages/extension-api/package.json (hunk 2 above)
Comment (elervik) [reply packages/extension-api/package.json:20]: Yes, accidental include. Something yarn adds?

File: packages/renderer/src/lib/pod/DeployPodToKube.svelte (hunk 1 above)
Comment (elervik) [reply packages/renderer/src/lib/pod/DeployPodToKube.svelte:46]: Forgive me, I dont know JS at all, but wouldnt setting it as a const, prevent the selector from binding a new value to it?

**Comment by benoitf** [comment]
Hello @elervik
Thanks for the enhancement

As Charlie previously mentioned, you need to SignOff your commit so the DCO check can be green.
https://github.com/containers/podman-desktop/pull/998/checks?check_run_id=9979768708

And you shouldn't remove the yarn.lock file, just keep it untouched.

**Comment by elervik** [comment]
> Hello @elervik Thanks for the enhancement
> 
> As Charlie previously mentioned, you need to SignOff your commit so the DCO check can be green. https://github.com/containers/podman-desktop/pull/998/checks?check_run_id=9979768708
> 
> And you shouldn't remove the yarn.lock file, just keep it untouched.

I rewrote the inital commit with a signature (verified), this is not enough? I dont understand how to change to sign unless you want the Author bit aswell, something like `git commit -m "feat: ..." -m "Author .........."`

Are you able to remove the file from the commit? If not, Ill close this PR and recreate it as I cant figure out how to do it from cli / gui.

**Comment by benoitf** [comment]
@elervik I can handle the removal of the file, but for the signed-off you need to do it

for example you can use `git commit -s` (or add --amend)

The commit in fact needs to include your email that is added with the -s flag using Signed-Off: comment

**Pull request closed without merging**